test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/SaveTempDataPropertyFilterTest.cs

```diff
@@ -113,6 +87,71 @@ public void SaveTempDataPropertyFilter_ReadsTempDataFromTempDataDictionary()
             Assert.Equal(0, controller.Test2);
         }
 
+        [Fact]
+        public void ApplyTempDataChanges_SetsPropertyValue()
+        {
+            // Arrange
+            var httpContext = new DefaultHttpContext();
+
+            var tempData = new TempDataDictionary(httpContext, Mock.Of<ITempDataProvider>())
+            {
+                { "TempDataProperty-Test", "Value" }
+            };
+            tempData.Save();
+
+            var controller = new TestControllerStrings()
+            {
+                TempData = tempData,
+            };
+
+            var provider = BuildSaveTempDataPropertyFilter(httpContext, tempData: tempData);
+            provider.PropertyHelpers = BuildPropertyHelpers<TestControllerStrings>();
+
+            // Act
+            provider.ApplyTempDataChanges(controller, httpContext);
+
+            // Assert
+            Assert.Equal("Value", controller.Test);
+            Assert.Null(controller.Test2);
+        }
+
+        private IList<PropertyHelper> BuildPropertyHelpers<TSubject>()
+        {
+            var subjectType = typeof(TSubject);
+
+            var properties = subjectType.GetProperties(
+                BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+
+            var result = new List<PropertyHelper>();
+
+            foreach (var property in properties)
+            {
+                result.Add(new PropertyHelper(property));
+            }
+
+            return result;
+        }
+
+        private SaveTempDataPropertyFilter BuildSaveTempDataPropertyFilter(
```

Comment:
Usually we call these kinds of methods `GetFoo...` or `CreateFoo...` for the sake of uniformity.